lib/smart_proxy_openscap/openscap_report_parser.rb

```diff
@@ -0,0 +1,81 @@
+# encoding=utf-8
+require 'openscap'
+require 'openscap/ds/arf'
+require 'openscap/xccdf/testresult'
+require 'openscap/xccdf/ruleresult'
+require 'openscap/xccdf/rule'
+require 'openscap/xccdf/fix'
+require 'openscap/xccdf/benchmark'
+require 'json'
+module Proxy::OpenSCAP
+  class Parse
+    def initialize(arf_data)
+      OpenSCAP.oscap_init
+      size         = arf_data.size
+      @arf_digest  = Digest::SHA256.hexdigest(arf_data)
+      @arf         = OpenSCAP::DS::Arf.new(:content => arf_data, :path => 'arf.xml.bz2', :length => size)
+      @results     = @arf.test_result.rr
+      sds          = @arf.report_request
+      bench_source = sds.select_checklist!
+      @bench       = OpenSCAP::Xccdf::Benchmark.new(bench_source)
+      @items       = @bench.items
+    end
+
+    def as_json
+      parse_report.to_json
+    end
+
+    private
+
+    def parse_report
+      report        = {}
+      report[:logs] = []
+      passed        = 0
+      failed        = 0
+      other         = 0
+      @results.each do |rr_id, result|
+        next if result.result == 'notapplicable' || result.result == 'notselected'
+        # get rules and their results
+        rule_data = @items[rr_id]
+        report[:logs] << populate_result_data(rr_id, result.result, rule_data)
+        # create metrics for the results
+        case result.result
+          when 'pass', 'fixed'
+            passed += 1
+          when 'fail'
+            failed += 1
+          else
+            other += 1
+        end
+      end
+      report[:digest]  = @arf_digest
+      report[:metrics] = { :passed => passed, :failed => failed, :other => other }
+      report
+    end
+
+    def populate_result_data(result_id, rule_result, rule_data)
+      log               = {}
+      log[:source]      = a8b(result_id)
+      log[:result]      = a8b(rule_result)
+      log[:title]       = a8b(rule_data.title)
+      log[:description] = a8b(rule_data.description)
+      log[:rationale]   = a8b(rule_data.rationale)
+      log[:references]  = hash_a8b(rule_data.references.map(&:to_hash))
```

Comment:
interestingly enough, `rule_data.references.map(&:to_hash)` returns an array of hashes so I get [this](https://gist.github.com/xprazak2/88a90d2a11159dc56d33) - note the :fixes node

something like:
```ruby
def hash_a8b(ary)
  a8b_array = []
  ary.each do |hash|
    a8b_hash = {}
    hash.each do |key, value|
      a8b_hash[a8b(key.to_s)] = a8b(value)
    end
    a8b_array << a8b_hash
  end
  a8b_array
end
```
works for me